File: apps/Lodgr.Api/Models/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("notifications")]
public class Notification
{
    [Key]
    [Column("notification_id")]
    public long NotificationId { get; set; }

    [Column("contract_id")]
    public long ContractId { get; set; }

    [Column("invoice_id")]
    public long? InvoiceId { get; set; }

    [Column("type")]
    public NotificationType Type { get; set; }

    [Required]
    [StringLength(255)]
    [Column("recipient_email")]
    public string RecipientEmail { get; set; } = string.Empty;

    [StringLength(255)]
    [Column("subject")]
    public string? Subject { get; set; }

    [Column("status")]
    public NotificationStatus Status { get; set; } = NotificationStatus.PENDING;

    [StringLength(100)]
    [Column("taskgate_job_id")]
    public string? TaskgateJobId { get; set; }

    [Column("sent_at")]
    public DateTime? SentAt { get; set; }

    [Column("error_message")]
    public string? ErrorMessage { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

public enum NotificationType
{
    INVOICE,
    PAYMENT_REMINDER,
    CONTRACT_EXPIRY,
    PAYMENT_CONFIRMED,
    WELCOME,
}

public enum NotificationStatus
{
    PENDING,
    SENT,
    FAILED,
}

File: apps/api/src/main/java/io/github/hoanghonghuy/lodgr/entity/Notification.java
package io.github.hoanghonghuy.lodgr.entity;

import io.github.hoanghonghuy.lodgr.enums.NotificationStatus;
import io.github.hoanghonghuy.lodgr.enums.NotificationType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter
@Setter
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notification_id")
    private Long notificationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "contract_id", nullable = false)
    private Contract contract;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "invoice_id")
    private Invoice invoice;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private NotificationType type;

    @Column(name = "recipient_email", nullable = false)
    private String recipientEmail;

    @Column(name = "subject")
    private String subject;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private NotificationStatus status;

    @Column(name = "taskgate_job_id")
    private String taskgateJobId;

    @Column(name = "sent_at")
    private LocalDateTime sentAt;

    @Column(name = "error_message")
    private String errorMessage;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
}
